compiler/rustc_middle/src/ty/adt.rs

```diff
@@ -103,6 +104,8 @@ pub struct AdtDefData {
     flags: AdtFlags,
     /// Repr options provided by the user.
     repr: ReprOptions,
+    /// Target features that functions taking objects of this type by value will enable.
```

Comment:
I'm not sure it actually makes sense to have this field on `AdtDefData`, it should always be equivalent to the query you added. This'll also let you avoid a bunch of the rmeta changes which seems like a win to me

Reply:
Good idea, the code is a lot simpler now :-)